inference-engine/src/auto_plugin/auto_plugin.cpp

```diff
@@ -24,13 +24,65 @@ namespace {
         return config;
     }
 
-    DeviceInformation SelectDevice(const std::vector<DeviceInformation>& metaDevices) {
+    std::string GetNetworkPrecision(const InferenceEngine::CNNNetwork &network) {
+        for (auto&& layer : network.getInputsInfo()) {
+            auto precision = layer.second->getPrecision();
+            auto name = std::string(precision.name());
+            if (name == "I8") {
+                name = "INT8";
+            }
+            return name;
+        }
+        return {};
+    }
```

Comment:
this is not network precision. It's precision of input layer..

Reply:
well, I remembered Maxim provided some code like 
`auto nGraphFunc = network.getFunction();
        for (auto &node : nGraphFunc->get_ordered_ops()) {
            // auto type0 = node->input_value(0).get_element_type(); //input
            auto type1 = node->input_value(1).get_element_type(); //weights
            std::cout << type1 << std::endl;
        }`
so is that expected? should we use typ1 or type0 as the network precision?

Reply:
int8 network - if we have FakeQuantize operations in the graph
fp16 network - if we have convolution weights stored in fp16
fp32 network - if we have convolution weights stored in fp32

Reply:
> > I'm a bit confused, as this PR overlaps heavily with (much larger) #5545, let's focus on 5545, OR change the 5545 to be merged into the branch from THIS requets.
> > When both requests are trying to merege to the master, we are reviewing the overlapping delta twice
> 
> ok, I will change the 5545 to be merged to this branch. the reason I create 5545 and cherry-pick this commit is because I want to test jenkins CI, it seems 5545 CI is OK, I will merge it to this. thanks!

Let's focus on #5545 , which already contains this PR's commit, I will close this PR later.